aiida_firecrest/transport.py

```diff
@@ -315,45 +564,75 @@ def copy(
         dereference: bool = False,
         recursive: bool = True,
     ) -> None:
+        """Copy a file or directory on the remote. FirecREST does not support symlink copying.
+
+        :param recursive: If True, copy directories recursively.
+        note that the non-recursive option is not implemented in FirecREST server.
+        And it's not used in upstream, anyways...
+
+        :param dereference: If True, copy the target of the symlink instead of the symlink itself.
+        """
+        # TODO: investigate overwrite (?)
+
         if not recursive:
             # TODO this appears to not actually be used upstream, so just remove there
             raise NotImplementedError("Non-recursive copy not implemented")
-        source = self._cwd.joinpath(remotesource).enable_cache()
-        destination = self._cwd.joinpath(remotedestination).enable_cache()
+        if dereference:
+            raise NotImplementedError(
+                "Dereferencing not implemented in FirecREST server"
+            )
+        source = self._cwd.joinpath(
+            remotesource
+        )  # .enable_cache() it's removed from from path.py to be investigated
```

Comment:
unclear, please consider to make an issue

Reply:
Previously, when `FcPath` was part of `aiida-firecrest` @chrisjsewell , in this way could control specifically if  cache should be enable or disable based on specific purpose. 
But when class  [got merged](https://github.com/eth-cscs/pyfirecrest/pull/43) into pyfirecrest, they took this `enable_cache()` away, in such a way that when you make a `joinpath()` the new path is just copying the flag from the original path. So here if `self._cwd` has `cache=True` then `joinpath()` returns an `FCPath` with `cache=True`, likewise for `cache=False`.

For now we have `self._cwd` with `cache=True` always. The thing that I'm not entirely sure about is, if this is going to mess things up when files on remote is changed...
In my manual tests, I haven't seen any wrong return, when files on remote has changed and yet `cache=True`.
But still this has to be investigated in a separate PR/issue.

Reply:
Alright, just opened an issue is opened on this:
https://github.com/aiidateam/aiida-firecrest/issues/41